scripts/extractSlideData.js

```diff
@@ -35,44 +35,53 @@ const extractSlideData = (folderName) => {
   try {
     const talksPath = path.resolve(folderName);
     const files = fs.readdirSync(talksPath);
+
+    const indexFilter = /index\.html$/;
     const htmlFilter = /\.html$/;
     const folderFilter = /\./;
 
-    let talks = files.filter((file) => {
+    let talks = [];
+    files.forEach((file) => {
       if (!folderFilter.test(file)) {
         const fileNames = [];
         listFilesInDirectory(path.join(talksPath, file), fileNames);
 
-        return fileNames.some((file) => htmlFilter.test(file));
+        return fileNames.forEach((file) => {
+          if (indexFilter.test(file)) {
+            talks.push(
+              file
+                .replace(talksPath, '')
+                .replace(/\\/g, '/')
+                .match(/(?<=\/).+$/)[0]
```

Comment:
Can you extract this regex out to after line 41? Also, can you add a comment (in the code) about what this does ?

Reply:
I have instead simplified it now, should be more readable. Can add a comment if necessary.

https://github.com/AhsanAyaz/slides/pull/18/commits/392ef97b0050cefa52069d7e5ead8d89e6a5296e